build-logic/build.gradle.kts

```diff
@@ -1,15 +1,18 @@
+buildscript {
+    dependencies {
+        with(libs.kotlin.gradle.get()) {
+            classpath("$group:$name:$embeddedKotlinVersion")
+        }
+    }
+}
```

Comment:
Forcing Gradle to use Kotlin 1.9 results in errors, which I think has to do with https://kotlinlang.org/docs/whatsnew19.html#separate-compiler-plugins-for-official-kotlin-libraries but in any case Gradle has always said not to force a different version of Kotlin for their runtime because they offer no guarantees ([Gradle 8.3 should have Kotlin 1.9 support](https://github.com/gradle/gradle/pull/25794)).

Reply:
ah so this overrides the version of kotlin uses when compiling the build script logic right?

Reply:
Correct

Reply:
https://docs.gradle.org/8.3-rc-1/release-notes.html

The rc releases sometimes have issues, but I've used them in the past with no problems. It should allow kapt to stay in if you really don't want to remove it yet.

If you do want to remove kapt, I have a stashed changeset that's most of the way there, and I can make a PR. Unless you want the cathartic experience of ripping it out 😅